hypothesis-python/src/hypothesis/searchstrategy/strategies.py

```diff
@@ -225,6 +225,12 @@ def recur_inner(other):
         accept.__name__ = name
         return property(accept)
 
+    def available(self, data):
+        """Returns whether this strategy can *currently* draw any
+        values.
```

Comment:
IMO this docstring should also mention *why* we need this method in addition to the `.is_empty` property, and explain the different semantics (i.e. that `self.available()` may change over time).

I would also like to document that this method should never draw any values, i.e. `data` is provided solely for access to its attributes.  This is the case in all current uses, but it took a little digging to establish!

Reply:
I've updated the PR to improve the comments as you suggested.

> I would also like to document that this method should never draw any values, i.e. data is provided solely for access to its attributes.

Are you sure ? currently `Bundle.available` call `data.draw(self_strategy)` (see https://github.com/HypothesisWorks/hypothesis/pull/2079/files#diff-a86f45cd52cce8094e34f2ae9b013a39R313)
I guess this is a kind of special draw (given it can only return a single constant value), but it's kind of confusing. So I've also added a comment there.